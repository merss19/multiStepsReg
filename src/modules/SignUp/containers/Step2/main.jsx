"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_redux_1 = require("react-redux");
var CustomSteps_1 = require("../CustomSteps");
var react_router_dom_1 = require("react-router-dom");
var classNames = require("classnames");
var storage_1 = require("tools/storage");
var const_1 = require("../../const");
var Button_1 = require("components/Button");
var CustomModal_1 = require("modules/CustomModal");
var ModalHOC_1 = require("components/HOC/ModalHOC");
var ducks_1 = require("./ducks");
var selectors_1 = require("../../selectors");
var mapStateToProps = function (state) { return ({
    programs: selectors_1.selectPrograms(state),
    choosenProgram: selectors_1.selectChoosenProgram(state),
    choosenPackageType: selectors_1.selectPackageType(state),
    packages: selectors_1.selectPackages(state),
    promo: selectors_1.selectPromo(state),
    currentPackage: selectors_1.selectPackage(state),
    promoError: selectors_1.selectPromoError(state)
}); };
var StepTwoComponent = /** @class */ (function (_super) {
    __extends(StepTwoComponent, _super);
    function StepTwoComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepTwoComponent.prototype.componentWillMount = function () {
        var _a = this.props, programs = _a.programs, getPrograms = _a.getPrograms, packages = _a.packages, setChoosenProgram = _a.setChoosenProgram, choosenProgram = _a.choosenProgram, getPackage = _a.getPackage;
        if (!packages) {
            getPackage();
        }
        if (!programs) {
            getPrograms();
        }
    };
    StepTwoComponent.prototype.componentWillReceiveProps = function (nextProps) {
        var _a = this.props, programs = _a.programs, choosenProgram = _a.choosenProgram, setChoosenProgram = _a.setChoosenProgram;
        if (nextProps.programs) {
            if (!choosenProgram) {
                if (nextProps.programs) {
                    nextProps.programs.map(function (program) {
                        if (program.get('id') % const_1.ProgramNum.count === const_1.ProgramNum.hero) {
                            setChoosenProgram(program.get('id'));
                        }
                    });
                }
            }
        }
    };
    StepTwoComponent.prototype.shouldComponentUpdate = function (nextProps) {
        return (nextProps.step !== this.props.step
            || nextProps.packages !== this.props.packages
            || nextProps.programs !== this.props.programs
            || nextProps.choosenProgram !== this.props.choosenProgram
            || nextProps.choosenPackageType !== this.props.choosenPackageType
            || nextProps.promo !== this.props.promo
            || nextProps.currentPackage !== this.props.currentPackage
            || nextProps.promoError !== this.props.promoError);
    };
    StepTwoComponent.prototype.getPromo = function () {
        var _a = this.props, getPackage = _a.getPackage, setPromo = _a.setPromo;
        if (!this.promoText || !this.promoText.value.length) {
            return;
        }
        getPackage(this.promoText.value);
        setPromo(this.promoText.value);
    };
    StepTwoComponent.prototype.paymentCreate = function () {
        var _this = this;
        var _a = this.props, choosenProgram = _a.choosenProgram, choosenPackageType = _a.choosenPackageType, paymentCreate = _a.paymentCreate, promo = _a.promo;
        var payload = {
            authToken: storage_1.storage.load('token'),
            data: {
                program: choosenProgram,
                package: choosenPackageType,
                promoName: promo
            }
        };
        paymentCreate(payload, function () { return _this.resultAction(); });
    };
    StepTwoComponent.prototype.resultAction = function () {
        var _a = this.props, changeStep = _a.changeStep, choosenProgram = _a.choosenProgram;
        storage_1.storage.save('program', choosenProgram, { path: '/', maxAge: 60 * 60 * 24 * 365 * 10 });
        changeStep(const_1.Steps.three);
    };
    StepTwoComponent.prototype.renderPrograms = function () {
        var _a = this.props, programs = _a.programs, setChoosenProgram = _a.setChoosenProgram;
        if (!programs) {
            return <div>Нет программ</div>;
        }
        else {
            return (programs.map(function (program) { return (<li key={program.get('id')} onClick={function () { setChoosenProgram(program.get('id')); }} className={classNames('options__item', {
                'is-active': program.get('isActive'),
                'g-hero': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.hero,
                'g-mather': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.mother,
                'g-extreme': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.extreme,
                'g-tomorrow': program.get('id') % const_1.ProgramNum.count == const_1.ProgramNum.tomorrow,
            })}>
            {program.get('name')}
          </li>); }));
        }
    };
    StepTwoComponent.prototype.renderPackages = function () {
        var _a = this.props, packages = _a.packages, setPackageType = _a.setPackageType;
        if (!packages) {
            return null;
        }
        return (packages.map(function (pckg) {
            return (<li key={pckg.get('id')} className={classNames('options__item', {
                'is-active': pckg.get('isActive')
            })} onClick={function () { return setPackageType(pckg.get('id')); }}>
            {pckg.get('name')}
          </li>);
        }));
    };
    StepTwoComponent.prototype.render = function () {
        var _this = this;
        var currentPackage = this.props.currentPackage;
        return (<div>
        {this.renderHeader()}
        <div className='entry__box'>
          <ul className='options options--white wide mb30'>
            {this.renderPrograms()}
          </ul>
          <ul className='options options--white options--types wide mt30'>
            {this.renderPackages()}
          </ul>
        </div>
        <div className='entry__box'>
          <div className='entry-program-price'>{currentPackage ? currentPackage.get('cost') + ' руб' : ''}</div>
          <div className='input--btn'>
            <input ref={function (input) { return _this.promoText = input; }} type='text' name='promo' className='input__field' placeholder='Eсть промокод, вводи'/>
            <Button_1.Button onClick={function () { return _this.getPromo(); }} type='button' styleBtn={Button_1.ButtonTypes.info}>
              Применить
            </Button_1.Button>
          </div>
          <Button_1.Button onClick={function () { return _this.paymentCreate(); }} type='button' styleBtn={Button_1.ButtonTypes.success} wide={true}>
            Далее
          </Button_1.Button>
        </div>

        <div className='entry__link text-center'>
          <div className='entry-nav__item'>
            <react_router_dom_1.Link to='/' onClick={function () { return _this.logOut(); }}>Выйти</react_router_dom_1.Link>
          </div>
        </div>
      </div>);
    };
    return StepTwoComponent;
}(CustomSteps_1.default));
exports.StepTwoComponent = StepTwoComponent;
exports.StepTwo = react_redux_1.connect(mapStateToProps, {
    setChoosenProgram: ducks_1.setChoosenProgram,
    setPackageType: ducks_1.setPackageType,
    toggleModal: CustomModal_1.toggleModal,
    paymentCreate: ducks_1.paymentCreate,
    setPromo: ducks_1.setPromo,
    getPackage: ducks_1.getPackage,
    getPrograms: ducks_1.getPrograms
})(StepTwoComponent);
exports.default = ModalHOC_1.modalHOC(exports.StepTwo, ducks_1.moduleName, ducks_1.subModuleName);
