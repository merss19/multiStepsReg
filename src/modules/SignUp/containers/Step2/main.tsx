import * as React from 'react';
import {connect} from 'react-redux';
import {FormProps} from 'redux-form';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';
import {Link} from 'react-router-dom';
import * as classNames from 'classnames';
import {storage} from 'tools/storage';
import {Steps, ProgramNum} from '../../const';
import {Program} from '../../interfaces';
import {Payment} from '../../interfaces';
import {Button, ButtonTypes} from 'components/Button';
import {toggleModal} from 'modules/CustomModal';
import {modalHOC} from 'components/HOC/ModalHOC';
import {
  setChoosenProgram,
  setPackageType,
  getPackage,
  getPrograms,
  setPromo,
  paymentCreate,
  moduleName,
  subModuleName
} from './ducks';
import {
  selectChoosenProgram,
  selectPromoError,
  selectPackageType,
  selectPackages,
  selectPrograms,
  selectPromo,
  selectPackage
} from '../../selectors';

interface OwnProps extends CustomStepsProps, FormProps<{}, {}, {}> {
  changeStep: (data: number) => void
  step: Steps
}

type ConnectedState = {
  choosenPackageType: number
  choosenProgram?: number
  packages: any
  currentPackage: any
  promo: string
  programs: Program[]
  promoError: string
}

type ConnectedDispatch = {
  setChoosenProgram(data: number): void
  setPackageType(data: number): void
  getPackage(data?: string): void
  getPrograms(): void
  paymentCreate(data: Payment, cb: () => void): void
  setPromo(data: string): void
  toggleModal(data: boolean): void
}

const mapStateToProps = (state: any): ConnectedState => ({
  programs: selectPrograms(state),
  choosenProgram: selectChoosenProgram(state),
  choosenPackageType: selectPackageType(state),
  packages: selectPackages(state),
  promo: selectPromo(state),
  currentPackage: selectPackage(state),
  promoError: selectPromoError(state)
});
type Props = OwnProps & ConnectedDispatch & ConnectedState

export class StepTwoComponent extends CustomSteps<Props> {

  promoText: any;

  componentWillMount() {
    const {programs, getPrograms, packages, setChoosenProgram, choosenProgram, getPackage} = this.props;
    if(!packages){
      getPackage();
    }
    if(!programs){
      getPrograms();
    }
  }
  componentWillReceiveProps(nextProps: Props){
    const {programs, choosenProgram, setChoosenProgram} = this.props
    if(nextProps.programs){
      if (!choosenProgram) {
        if(nextProps.programs){
          nextProps.programs.map((program: any) => {
            if (program.get('id') % ProgramNum.count === ProgramNum.hero) {
              setChoosenProgram(program.get('id'));
            }
          })
        }
      }
    }
  }
  shouldComponentUpdate(nextProps:Props){
    return (
      nextProps.step !== this.props.step
      ||  nextProps.packages !== this.props.packages
      ||  nextProps.programs !== this.props.programs
      ||  nextProps.choosenProgram !== this.props.choosenProgram
      ||  nextProps.choosenPackageType !== this.props.choosenPackageType
      ||  nextProps.promo !== this.props.promo
      ||  nextProps.currentPackage !== this.props.currentPackage
      ||  nextProps.promoError!== this.props.promoError
    );
  }

  getPromo() {
    const {getPackage, setPromo} = this.props;
    if (!this.promoText || !this.promoText.value.length) {
      return;
    }
    getPackage(this.promoText.value);
    setPromo(this.promoText.value);
  }

  paymentCreate() {
    const {choosenProgram, choosenPackageType, paymentCreate, promo} = this.props;
    let payload: Payment = {
      authToken: storage.load('token'),
      data: {
        program: choosenProgram,
        package: choosenPackageType,
        promoName:  promo
      }
    };
    paymentCreate(payload, () => this.resultAction());
  }

  resultAction() {
    const {changeStep, choosenProgram} = this.props;
    storage.save('program', choosenProgram,
      {path: '/', maxAge: 60 * 60 * 24 * 365 * 10});
    changeStep(Steps.three);
  }

  renderPrograms() {
    const {programs, setChoosenProgram} = this.props;
    if (!programs) {
      return <div>Нет программ</div>;
    } else {
      return (
        programs.map((program: any) => (
          <li key={program.get('id')}
              onClick={() =>{setChoosenProgram(program.get('id'))}}
              className={classNames('options__item', {
                'is-active': program.get('isActive'),
                'g-hero': program.get('id') % ProgramNum.count == ProgramNum.hero,
                'g-mather': program.get('id') % ProgramNum.count == ProgramNum.mother,
                'g-extreme': program.get('id') % ProgramNum.count == ProgramNum.extreme,
                'g-tomorrow': program.get('id') % ProgramNum.count == ProgramNum.tomorrow,
              })}
          >
            {program.get('name')}
          </li>)
        ))
    }
  }
  renderPackages() {
    const {packages, setPackageType} = this.props;
    if (!packages) {
      return null;
    }
    return (
      packages.map((pckg: any) => {
        return (
          <li
            key={pckg.get('id')}
            className={classNames('options__item', {
              'is-active': pckg.get('isActive')
            })}
            onClick={() => setPackageType(pckg.get('id'))}
          >
            {pckg.get('name')}
          </li>);
      })
    )
  }

  render() {
    const {currentPackage} = this.props;
    return (
      <div>
        {this.renderHeader()}
        <div className='entry__box'>
          <ul className='options options--white wide mb30'>
            {this.renderPrograms()}
          </ul>
          <ul className='options options--white options--types wide mt30'>
            {this.renderPackages()}
          </ul>
        </div>
        <div className='entry__box'>
          <div className='entry-program-price'>{currentPackage ? currentPackage.get('cost') + ' руб' : ''}</div>
          <div className='input--btn'>
            <input
              ref={input => this.promoText = input}
              type='text'
              name='promo'
              className='input__field'
              placeholder='Eсть промокод, вводи'
            />
            <Button
              onClick={() => this.getPromo()}
              type='button'
              styleBtn={ButtonTypes.info}
            >
              Применить
            </Button>
          </div>
          <Button
            onClick={() => this.paymentCreate()}
            type='button'
            styleBtn={ButtonTypes.success}
            wide={true}
          >
            Далее
          </Button>
        </div>

        <div className='entry__link text-center'>
          <div className='entry-nav__item'>
            <Link to='/' onClick={() => this.logOut()}>Выйти</Link>
          </div>
        </div>
      </div>);
  }
}

export const StepTwo = connect(
  mapStateToProps, {
    setChoosenProgram,
    setPackageType,
    toggleModal,
    paymentCreate,
    setPromo,
    getPackage,
    getPrograms
  })(StepTwoComponent);

export default modalHOC(StepTwo, moduleName, subModuleName);

