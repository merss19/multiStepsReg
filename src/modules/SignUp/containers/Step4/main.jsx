"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_router_dom_1 = require("react-router-dom");
var Button_1 = require("components/Button");
var CustomSteps_1 = require("../CustomSteps");
var StepFour = /** @class */ (function (_super) {
    __extends(StepFour, _super);
    function StepFour() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    StepFour.prototype.render = function () {
        var _this = this;
        return (<div className='entry__success'>
        <h3>Ваша заявка принята.</h3>
        <p>В ближайшее время мы с вами свяжемся!</p>
        <div className='text-center pt20'>
          <Button_1.Button type={'button'} styleBtn={Button_1.ButtonTypes.info}>
            <react_router_dom_1.Link to='/' onClick={function () { return _this.logOut(); }}>Вернуться на главный сайт</react_router_dom_1.Link>
          </Button_1.Button>
        </div>
      </div>);
    };
    return StepFour;
}(CustomSteps_1.default));
exports.StepFour = StepFour;
exports.default = StepFour;
