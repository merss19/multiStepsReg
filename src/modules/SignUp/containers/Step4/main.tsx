import * as React from 'react';
import {Link} from 'react-router-dom';
import {Button, ButtonTypes} from 'components/Button';
import CustomSteps, {CustomStepsProps} from '../CustomSteps';

interface OwnProps extends CustomStepsProps {}
export class StepFour extends CustomSteps<OwnProps> {
  render() {
    return (
      <div className='entry__success'>
        <h3>Ваша заявка принята.</h3>
        <p>В ближайшее время мы с вами свяжемся!</p>
        <div className='text-center pt20'>
          <Button
            type={'button'}
            styleBtn={ButtonTypes.info}
          >
            <Link to='/' onClick={() => this.logOut()}>Вернуться на главный сайт</Link>
          </Button>
        </div>
      </div>
    );
  }
}

export default StepFour;
