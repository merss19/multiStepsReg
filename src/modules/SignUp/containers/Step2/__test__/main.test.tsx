import {shallow, mount} from 'enzyme';
import {StepTwoComponent, StepTwo} from '../main';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import * as sinon from 'sinon';
import {Provider} from 'react-redux';
import configureStore from '../../../../../store';
import {BrowserRouter as Router} from 'react-router-dom';
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {storage} from '../../../../../tools/storage/index';
import {Steps} from '../../../const';
import {Button, ButtonTypes} from "../../../../../components/Button/button";
import * as Immutable from 'immutable';

const store = configureStore();
configure({adapter: new Adapter()});
let data = [
  {
    id: 1,
    name: 'hero',
    isActive: true
  },
  {
    id: 2,
    name: 'mother',
    isActive: false
  },
  {
    id: 3,
    name: 'extreme',
    isActive: false
  },
  {
    id: 4,
    name: 'tomorrow',
    isActive: false
  }
];

let changeStep,
  getPrograms,
  userCreate,
  setProgramName,
  handleSubmit,
  setToken,
  toggleModal,
  setChoosenProgram,
  setPackageType,
  getPackage,
  paymentCreate,
  setPromo,
  setError,
  isFetching,
  isLoad,
  isError,
  step,
  errMsg,
  programs,
  choosenPackageType,
  choosenProgram,
  packages,
  currentPackage,
  promo,
  promoError,
  signUpData,
  dataRes,
  match,
  location,
  callback,
  history;
const start = () => {
  beforeEach(() => {
    changeStep = jest.fn(),
      getPrograms = jest.fn(),
      callback = jest.fn(),
      paymentCreate = jest.fn((payload, callback) => callback())
    setProgramName = jest.fn(),
      setToken = jest.fn(),
      handleSubmit = jest.fn((callback) => callback()),
      toggleModal = jest.fn(),
      setError = jest.fn(),
      setChoosenProgram = jest.fn(() => true),
      setPackageType = jest.fn(),
      getPackage = jest.fn(),
      //paymentCreate = jest.fn(),
      setPromo = jest.fn(),
      isFetching = false,
      isLoad = false,
      isError = false,
      errMsg = 'error',
      step = Steps.one,
      choosenPackageType = 1,
      choosenProgram = 0,
      packages = Immutable.fromJS([
        {
          name: '1 человек',
          isActive: false,
          id: 1
        },
        {
          name: '2 человек',
          isActive: false,
          id: 2
        },
        {
          name: '1 ЧЕЛОВЕК + ФИТНЕС-БРАСЛЕТ',
          isActive: false,
          id: 3
        },
        {
          name: '1 ЧЕЛОВЕК + КОВРИК',
          isActive: false,
          id: 4
        },
        {
          name: '3 человек',
          isActive: false,
          id: 5
        },
      ]) ,
      currentPackage = Immutable.fromJS({
        cost: 3000,
        name: '1 человек',
        isActive: true,
        id: 1
      }),
      promo = '',
      promoError = '',
      programs = Immutable.fromJS(data),
      /*signUpData = {
        email: 'mail@mail.ru',
        password: 'qweqwe',
        passwordAgain: 'qweqwe',
        gender: 'male',
      },*/
      match = {
        isExact: true,
        params: {},
        path: "/task",
        url: "/task"
      },
      location = {},
      history = {
        replace: jest.fn()
      }
  });
  const props = {
    getPrograms: getPrograms,
    userCreate: userCreate,
    handleSubmit: handleSubmit,
    changeStep: changeStep,
    setProgramName: setProgramName,
    setToken: setToken,
    toggleModal: toggleModal,
    setError: setError,
    setChoosenProgram: setChoosenProgram,
    setPackageType: setPackageType,
    getPackage: getPackage,
    paymentCreate: paymentCreate,
    setPromo: setPromo,
    step: step,
    choosenPackageType: choosenPackageType,
    choosenProgram: choosenProgram,
    promo: promo,
    currentPackage: currentPackage,
    packages: packages,
    promoError: promoError,
    errMsg: errMsg,
    programs: programs,
    isFetching: isFetching,
    isLoad: isLoad,
    isError: isError,
    match: match,
    location: location,
    history: history
  };
  const buildComponent = () => {
    return shallow(<StepTwoComponent {...props}/>)
  };
  const buildComponentMount = () => {
    return mount(
      <Provider store={store}>
        <Router>
          <StepTwo {...props}/>
        </Router>
      </Provider>)
  };
  return {
    buildComponentMount,
    props,
    buildComponent
  }
};
describe('Component StepTwo', () => {
  let component;
  start();
  beforeEach(() => {
    component = start().buildComponent();
  });
  it('renders component', () => {
    expect(component).toHaveLength(1);
  });
  it('renders component Mount', () => {
    const component = start().buildComponentMount();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('getPackage', () => {
    expect(getPackage).toHaveBeenCalled()
    expect(setChoosenProgram).toHaveBeenCalledWith(1)
  });
  it('setChoosenProgram do not call', () => {
    choosenProgram = 6;
    component = start().buildComponent();
    //component.setProps({choosenProgram})
    expect(setChoosenProgram).toHaveBeenCalledWith(1)
  });
  it('Test componentWillMount', () => {
    expect(getPrograms).toHaveBeenCalled();
  });
  it('getPromo, promoText === null', () => {
    const component = start().buildComponent();
    const instance = component.instance() as StepTwoComponent;
    instance.getPromo();
    expect(instance.getPromo()).toBeUndefined()
  });
  it('getPromo, promoText !== null', () => {
    const component = start().buildComponent();
    const instance = component.instance() as StepTwoComponent;
    instance.promoText = {
      value: 'test'
    };
    instance.getPromo();
    expect(getPackage).toHaveBeenCalledWith('test', true)
    expect(setPromo).toHaveBeenCalledWith('test')
  });
  it('paymentCreate', () => {
    storage.save('token', 'test');
    const component = start().buildComponent();
    const instance = component.instance() as StepTwoComponent;
    instance.paymentCreate();
    const cb = sinon.spy();
    const payload = {
      authToken: 'test',
      data: {
        program: 0,
        package: 1,
      }
    }
    expect(paymentCreate).toHaveBeenCalled();
    //expect(paymentCreate).toHaveBeenCalledWith(payload, cb);
  });
  describe('Test shouldComponentUpdate', () => {
    it('step', () => {
      const instance = component.instance() as StepTwoComponent;
      step = Steps.two;
      let props = {
        ...start().props,
        step: step
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('packages', () => {
      const instance = component.instance() as StepTwoComponent;
      packages = Immutable.fromJS([
        {
          name: '1 человек',
          isActive: false,
          id: 1
        },
        {
          name: '2 человек',
          isActive: false,
          id: 2
        },
        {
          name: '1 ЧЕЛОВЕК + ФИТНЕС-БРАСЛЕТ',
          isActive: false,
          id: 3
        },
        {
          name: '1 ЧЕЛОВЕК + КОВРИК',
          isActive: false,
          id: 4
        }
      ]);
      let props = {
        ...start().props,
        packages: packages
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('choosenProgram', () => {
      const instance = component.instance() as StepTwoComponent;
      choosenProgram = 3;
      let props = {
        ...start().props,
        choosenProgram: choosenProgram
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('choosenPackageType', () => {
      const instance = component.instance() as StepTwoComponent;
      choosenPackageType = 3;
      let props = {
        ...start().props,
        choosenPackageType: choosenPackageType
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('promo', () => {
      const instance = component.instance() as StepTwoComponent;
      promo = 'tele2';
      let props = {
        ...start().props,
        promo: promo
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('promoError', () => {
      const instance = component.instance() as StepTwoComponent;
      promoError = 'tele2';
      let props = {
        ...start().props,
        promoError: promoError
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
    it('currentPackage', () => {
      const instance = component.instance() as StepTwoComponent;
      currentPackage = Immutable.fromJS({
        name: '2 человек',
        isActive: true,
        id: 2
      });
      let props = {
        ...start().props,
        currentPackage: currentPackage
      };
      const shouldUpdate = instance.shouldComponentUpdate(props);
      expect(shouldUpdate).toBe(true)
    });
  });
  it('call resultAction', () => {
    const instance = component.instance() as StepTwoComponent;
    instance.resultAction();
    expect(changeStep).toHaveBeenCalledWith(Steps.three);
    expect(parseInt(storage.load('program'))).toBe(choosenProgram);
    storage.remove('program');
  });
  it('call renderPackages', () => {
    const instance = component.instance() as StepTwoComponent;
    //console.log(component.debug())
    const li = component.find('ul').at(1).find('li').at(0);
    li.simulate('click');
    expect(setPackageType).toHaveBeenCalled();
    expect(component.find('ul').at(1).find('li')).toHaveLength(5);

  });
  it('call renderPrograms', () => {
    const instance = component.instance() as StepTwoComponent;
    //console.log(component.debug())
    expect(component.find('ul').at(0).find('li')).toHaveLength(4);
    expect(component.find('.is-active .g-hero').exists()).toBe(true);
    expect(component.find('.is-active .g-mather').exists()).toBe(false);
    expect(component.find('.is-active .g-extreme').exists()).toBe(false);
    expect(component.find('.is-active .g-tomorrow').exists()).toBe(false);
  });
  it('call renderPrograms, is-active - g-mather', () => {
    programs = programs.map(item => {
      if(item.get('id') === 1){
        return item.set('isActive', false)
      }
      if(item.get('id') === 2){
        return item.set('isActive', true)
      }
      return item;
    });
    const component = start().buildComponent();
    const li = component.find('.g-mather');
    li.simulate('click')
    expect(setChoosenProgram).toHaveBeenCalled();
    expect(component.find('.is-active .g-hero').exists()).toBe(false);
    expect(component.find('.is-active .g-mather').exists()).toBe(true);
    expect(component.find('.is-active .g-extreme').exists()).toBe(false);
    expect(component.find('.is-active .g-tomorrow').exists()).toBe(false);
  });
  it('call renderPrograms, is-active - g-extreme', () => {
    programs = programs.map(item => {
      if(item.get('id') === 1){
        return item.set('isActive', false)
      }
      if(item.get('id') === 3){
        return item.set('isActive', true)
      }
      return item;
    });
    const component = start().buildComponent();
    const li = component.find('.g-extreme');
    li.simulate('click')
    expect(setChoosenProgram).toHaveBeenCalled();
    expect(component.find('.is-active .g-hero').exists()).toBe(false);
    expect(component.find('.is-active .g-mather').exists()).toBe(false);
    expect(component.find('.is-active .g-extreme').exists()).toBe(true);
    expect(component.find('.is-active .g-tomorrow').exists()).toBe(false);
  });
  it('call renderPrograms, is-active - g-tomorrow', () => {
    programs = programs.map(item => {
      if(item.get('id') === 1){
        return item.set('isActive', false)
      }
      if(item.get('id') === 4){
        return item.set('isActive', true)
      }
      return item;
    });
    const component = start().buildComponent();
    const li = component.find('.g-tomorrow');
    li.simulate('click')
    expect(setChoosenProgram).toHaveBeenCalled();
    expect(component.find('.is-active .g-hero').exists()).toBe(false);
    expect(component.find('.is-active .g-mather').exists()).toBe(false);
    expect(component.find('.is-active .g-extreme').exists()).toBe(false);
    expect(component.find('.is-active .g-tomorrow').exists()).toBe(true);
  });
  it('call renderPrograms, programs === null', () => {
    programs = null
    const component = start().buildComponent();
    const instance = component.instance() as StepTwoComponent;
    expect(instance.renderPrograms()).toEqual(<div>Нет программ</div>);
  });
  describe('call render()', () => {
    it('call renderHeader', () => {
      const spy = sinon.spy(StepTwoComponent.prototype, 'renderHeader');
      const component = start().buildComponent();
      expect(spy.called).toBe(true);
    });
    it('call renderPackages', () => {
      const spy = sinon.spy(StepTwoComponent.prototype, 'renderPackages');
      const component = start().buildComponent();
      expect(spy.called).toBe(true);
    });
    it('call renderPrograms', () => {
      const spy = sinon.spy(StepTwoComponent.prototype, 'renderPrograms');
      const component = start().buildComponent();
      expect(spy.called).toBe(true);
    });
    it('render currentPackage', () => {
      const component = start().buildComponent();
      const price = component.find('.entry-program-price').text();
      expect(price).toBe(start().props.currentPackage.get('cost') + ' руб');
    });
    it('test input promo', () => {
      // const component = start().buildComponentMount();
      // const instance = component.instance() as StepTwoComponent;
      //const promo = component.find('input[name="promo"]');
      //console.log(component.debug())
      //console.log('promoooo')
      //console.log(component.find('input[name="promo"]').instance())
      //promo.simulate('focus');
      // promo.simulate('change', {target: {value: 'testPromo'}});
      // promo.simulate('blur');
      // console.log(instance.promoText)
      //expect(promo.props().value).toBe('testPromo');
    });
    it('render promo button', () => {
      const button = component.find(Button).at(0)
      expect(button.exists()).toBe(true);
      expect(button.props().type).toBe('button');
      expect(button.props().styleBtn).toBe(ButtonTypes.info);
      expect(button.props().wide).toBe(false);
    });
    it('click promo Button', () => {
      const spy = sinon.spy(StepTwoComponent.prototype, 'getPromo');
      const button = component.find(Button).at(0);
      button.simulate('click');
      expect(spy.called).toBe(true);
    });
    it('render next button', () => {
      const button = component.find(Button).at(1)
      expect(button.exists()).toBe(true);
      expect(button.props().type).toBe('button');
      expect(button.props().styleBtn).toBe(ButtonTypes.success);
      expect(button.props().wide).toBe(true);
    });
    it('click next Button', () => {
      const spy = sinon.spy(StepTwoComponent.prototype, 'paymentCreate');
      const button = component.find(Button).at(1);
      button.simulate('click');
      expect(spy.called).toBe(true);
    });
    it('click Link', () => {
      const spy = sinon.spy(StepTwoComponent.prototype, 'logOut');
      const button = component.find('Link');
      button.simulate('click');
      expect(spy.called).toBe(true);
    });
  })
});


