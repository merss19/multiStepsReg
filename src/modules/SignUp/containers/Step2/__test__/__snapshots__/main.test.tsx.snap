// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StepTwo renders correctly 1`] = `
<div>
  <div
    className="entry__header"
  >
    <h2
      className="entry__title entry__title--auth text-center "
    >
      Регистрация
    </h2>
  </div>
  <div
    className="entry__box"
  >
    <ul
      className="options options--white wide mb30"
    >
      <li
        className="options__item is-active g-hero"
        key="1"
        onClick={[Function]}
      >
        hero
      </li>
      <li
        className="options__item g-mather"
        key="2"
        onClick={[Function]}
      >
        mother
      </li>
      <li
        className="options__item g-extreme"
        key="3"
        onClick={[Function]}
      >
        extreme
      </li>
      <li
        className="options__item g-tomorrow"
        key="4"
        onClick={[Function]}
      >
        tomorrow
      </li>
    </ul>
    <ul
      className="options options--white options--types wide mt30"
    >
      <li
        className="options__item"
        key="1"
        onClick={[Function]}
      >
        1 человек
      </li>
      <li
        className="options__item"
        key="2"
        onClick={[Function]}
      >
        2 человек
      </li>
      <li
        className="options__item"
        key="3"
        onClick={[Function]}
      >
        1 ЧЕЛОВЕК + ФИТНЕС-БРАСЛЕТ
      </li>
      <li
        className="options__item"
        key="4"
        onClick={[Function]}
      >
        1 ЧЕЛОВЕК + КОВРИК
      </li>
      <li
        className="options__item"
        key="5"
        onClick={[Function]}
      >
        3 человек
      </li>
    </ul>
  </div>
  <div
    className="entry__box"
  >
    <div
      className="entry-program-price"
    >
      3000 руб
    </div>
    <div
      className="input--btn"
    >
      <input
        className="input__field"
        name="promo"
        placeholder="Eсть промокод, вводи"
        type="text"
      />
      <Component
        onClick={[Function]}
        prefix="btn"
        styleBtn="secondary"
        type="button"
        wide={false}
      >
        Применить
      </Component>
    </div>
    <Component
      onClick={[Function]}
      prefix="btn"
      styleBtn="primary"
      type="button"
      wide={true}
    >
      Далее
    </Component>
  </div>
  <div
    className="entry__link text-center"
  >
    <div
      className="entry-nav__item"
    >
      <Link
        onClick={[Function]}
        replace={false}
        to="/"
      >
        Выйти
      </Link>
    </div>
  </div>
</div>
`;
